core/src/utilities/helpers/global-search-helpers.js

```diff
@@ -1,13 +1,195 @@
+      if (GenericHelpers.isFunction(this.search.searchProvider.onEnter) && keyCode === KEYCODE_ENTER) {
+        this.search.searchProvider.onEnter();
+      } else if (GenericHelpers.isFunction(this.search.searchProvider.onEscape) && keyCode === KEYCODE_ESC) {
+        this.search.searchProvider.onEscape();
+      } else if (keyCode === KEYCODE_ARROW_DOWN) {
+        if (displaySearchResult) {
+          document.querySelector('.luigi-search-result-item__0').childNodes[0].setAttribute('aria-selected', 'true');
+          document.querySelector('.luigi-search-result-item__0').focus();
+        }
+      } else if (GenericHelpers.isFunction(this.search.searchProvider.onInput)) {
+        this.search.searchProvider.onInput();
+      }
+    } else {
+      console.warn('GlobalSearch is not available.');
+    }
+  }
+
+  calcSearchResultItemSelected(direction) {
+    let renderedSearchResultItems = this.luigiCustomSearchItemRenderer__slotContainer.children;
+    if (renderedSearchResultItems) {
+      for (let index = 0; index < renderedSearchResultItems.length; index++) {
+        let { childNodes, nextSibling, previousSibling } = renderedSearchResultItems[index];
+        let nodeSibling;
+        if (childNodes[0].getAttribute('aria-selected') === 'true') {
+          if (direction === KEYCODE_ARROW_DOWN) {
+            nodeSibling = nextSibling !== null ? nextSibling : renderedSearchResultItems[0];
+          }
+          if (direction === KEYCODE_ARROW_UP) {
+            nodeSibling =
+              previousSibling !== null
+                ? previousSibling
+                : renderedSearchResultItems[renderedSearchResultItems.length - 1];
+          }
+          childNodes[0].setAttribute('aria-selected', 'false');
+          nodeSibling.childNodes[0].setAttribute('aria-selected', 'true');
+          nodeSibling.focus();
+          break;
+        }
+      }
+    }
+  }
+
+  clearAriaSelected() {
+    let renderedSearchResultItems = this.luigiCustomSearchItemRenderer__slotContainer.children;
+    if (renderedSearchResultItems) {
+      for (let index = 0; index < renderedSearchResultItems.length; index++) {
+        let element = renderedSearchResultItems[index];
+        if (element.childNodes[0].getAttribute('aria-selected') === 'true') {
+          element.childNodes[0].setAttribute('aria-selected', 'false');
+        }
+      }
+    }
+  }
+
+  closeSearchResult() {
+    this.dispatch('closeSearchResult');
+  }
+
+  onSearchResultItemSelected(searchResultItem) {
+    if (this.search && GenericHelpers.isFunction(this.search.searchProvider.onSearchResultItemSelected)) {
+      this.search.searchProvider.onSearchResultItemSelected(searchResultItem);
+    } else if (GenericHelpers.isFunction(this.search.searchProvider.onEscape) && event.keyCode === KEYCODE_ESC) {
+      this.search.searchProvider.onEscape();
+    }
+  }
+
+  handleKeydown(result, { keyCode }, inputElement, luigiCustomSearchItemRenderer__slotContainer) {
```

Comment:
Some alternative name for 'luigiCustomSearchItemRenderer__slotContainer' argument? Perhaps 'slotContainer' is good enough?

Reply:
To me original variable name is better, as it also provides the information what type of renderer is used.

Reply:
I agree with both sites :)
Indeed this variable is a bit long and also the underscore is maybe outdated. On the other hand when you only read 'slotContainer‘ you don’t know what the variable is standing for.
What we definitely do not need is 'luigi' because it is obvious and just an internal variable and we can somehow get rid of the underscore.
If 'customSearchItemRendererSlotContainer' is a better name... i don't know either :)

Reply:
OK - let's leave it as is. In the future we can also consider config object instead of multiple args to make it more robust :)

Reply:
Changed:
`luigiCustomSearchItemRenderer__slotContainer` -> `customSearchItemRendererSlotContainer`
`luigiCustomSearchItemRenderer__slot` - `customSearchItemRendererSlot`